targets/TARGET_STM/CMakeLists.txt

```diff
@@ -1,42 +1,29 @@
 # Copyright (c) 2020 ARM Limited. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
 
-if("STM32F0" IN_LIST MBED_TARGET_LABELS)
-    add_subdirectory(TARGET_STM32F0)
-elseif("STM32F1" IN_LIST MBED_TARGET_LABELS)
-    add_subdirectory(TARGET_STM32F1)
-elseif("STM32F2" IN_LIST MBED_TARGET_LABELS)
-    add_subdirectory(TARGET_STM32F2)
-elseif("STM32F3" IN_LIST MBED_TARGET_LABELS)
-    add_subdirectory(TARGET_STM32F3)
-elseif("STM32F4" IN_LIST MBED_TARGET_LABELS)
-    add_subdirectory(TARGET_STM32F4)
-elseif("STM32F7" IN_LIST MBED_TARGET_LABELS)
-    add_subdirectory(TARGET_STM32F7)
-elseif("STM32G0" IN_LIST MBED_TARGET_LABELS)
-    add_subdirectory(TARGET_STM32G0)
-elseif("STM32G4" IN_LIST MBED_TARGET_LABELS)
-    add_subdirectory(TARGET_STM32G4)
-elseif("STM32H7" IN_LIST MBED_TARGET_LABELS)
-    add_subdirectory(TARGET_STM32H7)
-elseif("STM32L0" IN_LIST MBED_TARGET_LABELS)
-    add_subdirectory(TARGET_STM32L0)
-elseif("STM32L1" IN_LIST MBED_TARGET_LABELS)
-    add_subdirectory(TARGET_STM32L1)
-elseif("STM32L4" IN_LIST MBED_TARGET_LABELS)
-    add_subdirectory(TARGET_STM32L4)
-elseif("STM32L5" IN_LIST MBED_TARGET_LABELS)
-    add_subdirectory(TARGET_STM32L5)
-elseif("STM32WB" IN_LIST MBED_TARGET_LABELS)
-    add_subdirectory(TARGET_STM32WB)
-endif()
+add_subdirectory(TARGET_STM32F0 EXCLUDE_FROM_ALL)
```

Comment:
I guess if a subdirectory happened to add multiple targets they would be all excluded from the ALL target using this approach. However, the main reason is because this is the approach we've taken in other places in Mbed. Is there a good reason to change it?

Reply:
Excluding adding the target is the semantically correct thing to do. We want to add all the directories unconditionally, to learn about all their targets. Then, we want to select which target to build/link with. We've so far in Mbed OS decided not to build all targets (CMake default behavior) when the ALL target is built, but instead to build only what isn't explicitly excluded (which is everything except the app). To fix this, we would remove EXCLUDE_FROM_ALL in Mbed OS, and specify which target to build (either from Mbed CLI 2 or directly when using CMake). As a side benefit, having an ALL target helps us to test building all possible board targets easily (just build the ALL target).

Reply:
Fixing the ALL target is out of the scope of this PR, though.

Reply:
Please create a story for this work so we can work on this.

Reply:
Yes, I was aware of the broken ALL target. However, I don't think using `EXCLUDE_FROM_ALL` to exclude the library targets individually is semantically correct either: we're misusing `EXCLUDE_FROM_ALL` currently as a workaround for our misconfigured ALL target and because `mbed-tools` invokes CMake poorly (it always tries to build ALL). So, moving `EXCLUDE_FROM_ALL` from `add_subdirectory` to `add_library` when we're just going to remove it altogether later seems unnecessary to me.

Reply:
Agreed